ui/src/shared/components/SourceIndicator.tsx

```diff
@@ -0,0 +1,50 @@
+import React, {SFC} from 'react'
+import PropTypes from 'prop-types'
+import _ from 'lodash'
+import uuid from 'uuid'
+
+import ReactTooltip from 'react-tooltip'
+
+import {Source} from 'src/types'
+
+interface Props {
+  sourceOverride?: Source
+}
+
+const SourceIndicator: SFC<Props> = (
+  {sourceOverride},
+  {source: {name, url}}
+) => {
+  const sourceName: string = _.get(sourceOverride, 'name', name)
+  const sourceUrl: string = _.get(sourceOverride, 'url', url)
+
+  if (sourceName) {
```

Comment:
in this case i find the return of null below a bit confusing -- maybe a ternary here would make this feel more familiar?